From c89817a4e7b9d0ab50790e0b77e5ede6f061619e Mon Sep 17 00:00:00 2001
From: Pascal Paillet <p.paillet@foss.st.com>
Date: Mon, 10 Mar 2025 18:05:04 +0100
Subject: [PATCH 1656/1695] product/optee-stm32mp1: fix default latency value

Fix default OPP change latency value to 1000uS.
Zero is treated as infinite value by linux SCMI driver:
cpu/cpu0/cpufreq/cpuinfo_transition_latency:4294967295

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 7d8b17d51eae07b4e331ea05af8fd51e70040fa4 product/optee-stm32mp1: fix default latency value]
Change-Id: Ia7f9781eeb3407f29cc4f759a352d1f84ca22a18
Signed-off-by: Pascal Paillet <p.paillet@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/SCP-firmware/+/443250
Reviewed-by: Pascal PAILLET <p.paillet@st.com>
Tested-by: Pascal PAILLET <p.paillet@st.com>
Domain-Review: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
Reviewed-by: Valentin CARON <valentin.caron@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
---
 .../SCP-firmware/product/optee-stm32mp1/fw/config_all.c         | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/core/lib/scmi-server/SCP-firmware/product/optee-stm32mp1/fw/config_all.c b/core/lib/scmi-server/SCP-firmware/product/optee-stm32mp1/fw/config_all.c
index 56fc148cf..41877581d 100644
--- a/core/lib/scmi-server/SCP-firmware/product/optee-stm32mp1/fw/config_all.c
+++ b/core/lib/scmi-server/SCP-firmware/product/optee-stm32mp1/fw/config_all.c
@@ -736,7 +736,7 @@ static void set_resources(struct scpfw_config *cfg)
                         (fwk_id_t)FWK_ID_ELEMENT_INIT(FWK_MODULE_IDX_PSU, psu_index);
                     dvfs_data[dvfs_index].clock_id =
                         (fwk_id_t)FWK_ID_ELEMENT_INIT(FWK_MODULE_IDX_CLOCK, clock_index);
-                    dvfs_data[dvfs_index].latency = 0; /* not set, used for async access */
+                    dvfs_data[dvfs_index].latency = 1000;
                     dvfs_data[dvfs_index].sustained_idx = perfd_cfg->initial_opp;
                     dvfs_data[dvfs_index].opps =
                         fwk_mm_calloc(perfd_cfg->dvfs_opp_count,
-- 
2.47.2

